### PR TITLE
compatibility for elgg 2.1 and extended_tinymce

### PR DESCRIPTION
**Pull request by propertunist**
this version of the code includes functions to grab the cursor's absolute location in tinymce (tinymce_extended) and so places the popup menu in the correct location. i have also fixed other bugs relating to usernames with hyphens and lookups with full stops in them.
this does not yet appear to work with plaintext fields and i haven't put any specific code in to improve that aspect. i have not tested ckeditor at all.

### CODE REVIEW

File: start.php [hunk 1]
```diff
@@ -33,12 +30,14 @@ function mentions_get_regex() {
...
 	// preceeded by @, and possibly surrounded by word boundaries.
-	return '/[\b]?@([\p{L}\p{M}_\.0-9]+)[\b]?/iu';
+	return '/[\b]?@([\p{L}\p{M}-_\.0-9]+)[\b]?/iu';
 }
 
 function mentions_get_views() {
 	// allow plugins to add additional views to be processed for usernames
 	$views = array('output/longtext');
+	//, 'river/item');
+	//,'object/thewire');
```
Comment (juho-jaakkola) [inline start.php]: What are these comments?

File: start.php [hunk 2]
```diff
@@ -54,8 +53,8 @@ function mentions_get_views() {
  * @return string
  */
 function mentions_rewrite($hook, $entity_type, $returnvalue, $params) {
-
 	$regexp = mentions_get_regex();
+
```
Comment (juho-jaakkola) [inline start.php]: Please avoid whitespace changes what are not related to the actual functional changes.

File: views/default/js/mentions/autocomplete.js [hunk 3]
```diff
@@ -11,6 +11,49 @@ define(function(require) {
 	var elgg = require('elgg');
 	var callback;
 
+ /**
+  *  positionMentionPopup()
```
Comment (juho-jaakkola) [inline views/default/js/mentions/autocomplete.js]: Indentation is wrong.

File: views/default/js/mentions/autocomplete.js [hunk 4]
```diff
@@ -11,6 +11,49 @@ define(function(require) {
...
+	*
+	*  get position in pixels of the editor's cursor and place the mention popup relative to it.
+	**/
+
+	var positionMentionPopup = function() {
+		if (typeof tinyMCE.activeEditor != "undefined")
```
Comment (juho-jaakkola) [inline views/default/js/mentions/autocomplete.js]: All editor specific code should go to one of these files: https://github.com/Elgg/mentions/tree/master/views/default/js/mentions/editors

We do not want to load tinymce specific code on every page load in sites that aren't even using tinymce.

**Comment by juho-jaakkola** [comment]
The recommended way is to make a separate branch for each individual change. That way you can also make a separate PR for each of them. Small PRs are easier and quicker to review (and fix if necessary) than PRs with multiple commits. Therefore they can also get accepted faster.

File: start.php (hunk 1 above)
Comment (propertunist) [inline start.php]: i was in the process of extending the application of the mentions plugin to apply to every location in elgg where it makes sense to have mentions visible, such as river items and wire posts. however, i found that it was interfered with by wire_tools on my installation so i commented them out. i didn't fully prepare the code for a straight integration into the parent version of mentions before creating the pull request. if you'd prefer me to do that then i will have to cancel this one and do it again in a few days.

File: start.php (hunk 2 above)
Comment (propertunist) [inline start.php]: i did not make these changes thinking they would be pulled in to the main code, so i coded them according to my own preference.

**Comment by jeabakker** [comment]
sorry it took too long to get to this
